### PR TITLE
DB-10059 Implement SQL procedure that allow setting retention period

### PR DESCRIPTION
**Pull request by hatyo**

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java [hunk 1]
```diff
@@ -1090,6 +1090,21 @@ public static String getSqlConglomsInSchema(){
         return sqlConglomsInSchema;
     }
 
+    private static final String sqlGetTablesInSchema= "SELECT TABLEID FROM SYS.SYSTABLES WHERE SCHEMAID = ?";
```
[the remark this answers is not in the document]
Comment (hatyo) [reply splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java:1093]: done.

File: splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java [hunk 2]
```diff
@@ -1090,6 +1090,21 @@ public static String getSqlConglomsInSchema(){
...
+    private static List<TableDescriptor> getTablesInSchema(DataDictionary dataDictionary,
+                                                           Connection connection,
+                                                           String schemaId) throws SQLException, StandardException {
+        PreparedStatement statement = connection.prepareStatement(sqlGetTablesInSchema);
+        statement.setString(1, schemaId);
+        ResultSet resultSet = statement.executeQuery();
+        List<TableDescriptor> tableDescriptors = new ArrayList<>();
+        while(resultSet.next()) {
+            tableDescriptors.add(dataDictionary.getTableDescriptor(new BasicUUID(resultSet.getString(1))));
+        }
```
[the remark this answers is not in the document]
Comment (hatyo) [reply splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java:1104]: done.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceSystemProcedures.java [hunk 3]
```diff
@@ -1186,6 +1186,16 @@ public SpliceSystemProcedures(DataDictionary dictionary) {
                             .build();
                     procedures.add(purgeDeletedRows);
 
+                    Procedure minRetentionPeriod = Procedure.newBuilder().name("SET_MIN_RETENTION_PERIOD")
+                            .varchar("schemaName", 128)
+                            .varchar("tableName", 128)
```
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceSystemProcedures.java:1191]: you may use .catalog("tableName") or .catalog("schemaName")

**Review by jyuanca — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**